### ISSUE TITLE
[bitnami/kafka] Pod failed to start after configuring SASL/PLAIN for bitnami/kafka

### ISSUE THREAD

**Issue by zibuyule** [bitnami/charts#17965]
### Name and Version

bitnami/kafka --version 23.0.6

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1. replicaCount: 3 
2. use storageclass： nfs-csi
3. After completing the above two steps , The pod will be successfully created; and you can see that the kafka cluster is working normally.
4. Once I added the following configuration to values.yaml With SASL/PLAIN enabled, the pod will fail to start.

Post my values.yaml configuration as follows:

```
authorizerClassName: kafka.security.authorizer.AclAuthorizer

auth:
  clientProtocol: sasl
  externalClientProtocol: sasl
  interBrokerProtocol: plaintext
  sasl:
    mechanisms: plain,scram-sha-256,scram-sha-512
    interBrokerMechanism: plain
    jaas:
      clientUsers:
        - yourusername
      clientPasswords:
        - yourpassword
```

![kafka-error](https://github.com/bitnami/charts/assets/5792519/58d62fea-d4ea-448a-ab86-fab224eba957)

### Are you using any custom parameters or values?

Once I added the following configuration to values.yaml With SASL/PLAIN enabled, the pod will fail to start.

Post my values.yaml configuration as follows:

```
authorizerClassName: kafka.security.authorizer.AclAuthorizer

auth:
  clientProtocol: sasl
  externalClientProtocol: sasl
  interBrokerProtocol: plaintext
  sasl:
    mechanisms: plain,scram-sha-256,scram-sha-512
    interBrokerMechanism: plain
    jaas:
      clientUsers:
        - yourusername
      clientPasswords:
        - yourpassword
```

### What is the expected behavior?

The pod can start normally; the kafka cluster can work normally

### What do you see instead?

```
(base) root@infra-master1-xxxx:~/kafka-bitnami/kafka# kubectl get all -n kafka
NAME          READY   STATUS             RESTARTS     AGE
pod/kafka-0   0/1     CrashLoopBackOff   1 (2s ago)   9s
pod/kafka-1   0/1     CrashLoopBackOff   1 (3s ago)   9s
pod/kafka-2   0/1     CrashLoopBackOff   1 (3s ago)   9s

NAME                     TYPE           CLUSTER-IP      EXTERNAL-IP    PORT(S)                      AGE
service/kafka            LoadBalancer   10.233.32.178   10.86.26.245   9092:30325/TCP               9s
service/kafka-headless   ClusterIP      None            <none>         9092/TCP,9094/TCP,9093/TCP   9s

NAME                     READY   AGE
statefulset.apps/kafka   0/3     9s

(base) root@infra-master1-xxxx:~/kafka-bitnami/kafka# kubectl logs -f kafka-0 -n kafka
Defaulted container "kafka" out of: kafka, volume-permissions (init)
kafka 13:39:44.39 
kafka 13:39:44.39 Welcome to the Bitnami kafka container
kafka 13:39:44.39 Subscribe to project updates by watching https://github.com/bitnami/containers
kafka 13:39:44.40 Submit issues and feature requests at https://github.com/bitnami/containers/issues
kafka 13:39:44.40 
kafka 13:39:44.40 INFO  ==> ** Starting Kafka setup **
kafka 13:39:44.45 WARN  ==> You set the environment variable ALLOW_PLAINTEXT_LISTENER=yes. For safety reasons, do not use this flag in a production environment.
kafka 13:39:44.45 INFO  ==> Initializing Kafka...
kafka 13:39:44.46 INFO  ==> No injected configuration files found, creating default config files
kafka 13:39:44.83 INFO  ==> Configuring Kafka for inter-broker communications with PLAINTEXT authentication.
kafka 13:39:44.84 WARN  ==> Inter-broker communications are configured as PLAINTEXT. This is not safe for production environments.
kafka 13:39:44.84 INFO  ==> Configuring Kafka for client communications with SASL_PLAINTEXT authentication.
kafka 13:39:44.85 INFO  ==> Generating JAAS authentication file
kafka 13:39:44.86 INFO  ==> Creating users in Zookeeper
```

**Comment by javsalgar** [comment]
Hi!

Could you set `image.debug=true` to see if we get more insight on the issue?

**Comment by zibuyule** [comment]
Hi!

Attached here is my screenshot.

![debug-kafka](https://github.com/bitnami/charts/assets/5792519/ab8a1c1f-baff-496c-97e1-76a2d3f1b517)

**Comment by javsalgar** [comment]
Hi,

Could you check if the changes done in https://github.com/bitnami/charts/pull/17507 fix the issue?

**Comment by danielvolkov** [comment]
have the same issue chart version  23.0.7

**Comment by joancafom** [comment]
Hi! As mentioned by my colleague, we recently introduced major changes in the chart. This new version includes changes and fixes regarding SASL, so it might probably address this issue as well. Could you please check if you experience the same with the latest available release `25.1.2`.

**Comment by zibuyule** [comment]
> Hi! As mentioned by my colleague, we recently introduced major changes in the chart. This new version includes changes and fixes regarding SASL, so it might probably address this issue as well. Could you please check if you experience the same with the latest available release `25.1.2`.

I used the latest chart a few days ago. It has been successful. Thanks.
